pkg/ddl/ddl.go

```diff
@@ -416,20 +449,51 @@ func (sv *schemaVersionManager) setSchemaVersion(job *model.Job, store kv.Storag
 }
 
 // lockSchemaVersion gets the lock to prevent the schema version from being updated.
-func (sv *schemaVersionManager) lockSchemaVersion(jobID int64) {
+func (sv *schemaVersionManager) lockSchemaVersion(jobID int64) error {
 	ownerID := sv.lockOwner.Load()
 	// There may exist one job update schema version many times in multiple-schema-change, so we do not lock here again
 	// if they are the same job.
 	if ownerID != jobID {
 		sv.schemaVersionMu.Lock()
 		sv.lockOwner.Store(jobID)
+		if sv.etcdClient != nil && variable.DDLVersion.Load() == model.TiDBDDLV2 {
+			se, err := concurrency.NewSession(sv.etcdClient)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			mu := concurrency.NewMutex(se, ddlSchemaVersionKeyLock)
+			if err := mu.Lock(sv.ctx); err != nil {
+				return errors.Trace(err)
+			}
+			sv.seMaps[jobID] = se
+			sv.lockMaps[jobID] = mu
+		}
 	}
+	return nil
 }
 
 // unlockSchemaVersion releases the lock.
 func (sv *schemaVersionManager) unlockSchemaVersion(jobID int64) {
 	ownerID := sv.lockOwner.Load()
 	if ownerID == jobID {
+		if sv.etcdClient != nil && variable.DDLVersion.Load() == model.TiDBDDLV2 {
+			se, ok := sv.seMaps[jobID]
+			if !ok {
+				return
+			}
+			mu, ok := sv.lockMaps[jobID]
```

Comment:
Could we put `seMaps` and `lockMaps` together? Then we only need to do this check and delete once. like `lockInfoMap map[int64]*lockInfo`